react-app/src/components/InputText/InputText.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import FontAwesomeIcon from '@fortawesome/react-fontawesome';
+import { faEye, faEyeSlash } from '@fortawesome/fontawesome-free-solid';
+
+import './InputText.css';
+
+export default class InputText extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: this.props.type,
+      togglePasswordVisibilityButtonIcon: faEye
+    };
+
+    this.togglePasswordVisibility = this.togglePasswordVisibility.bind(this);
+  }
+
+  togglePasswordVisibility() {
```

Comment:
If both states switch at the same time there's no reason to make them both part of the state, we could have just the `type` state and calculate the icon needed when we need to use it.